Automattic/wp-calypso
# My Email: Add permission error view

#### Changes proposed in this Pull Request

* Adds a no-permission view when visiting the Inbox without the appropriate permissions
* Uses hooks and functional equivalents to pull in state logic for consistency with other units
* Renames the unit to *.jsx since React/JSX semantics are used.

#### Testing instructions



* Add an editor to your site
* Visit the `/inbox`
* Confirm that you are greeted with an empty content screen with a permissions error


<img width="900" alt="Screenshot 2021-10-05 at 1 16 27 AM" src="https://user-images.githubusercontent.com/277661/135941264-b615b00c-5fcc-4268-ad87-07644a2e6f50.png">

* Confirm that the rest of the inbox works as expected.



Related to D67803-code
